src/main/scala/cromwell/binding/command/ParameterCommandPart.scala

```diff
@@ -1,105 +1,50 @@
- * @param postfixQuantifier - The `?`, `*`, or `+` after the variable name.  This means "optional", "0-or-more", and "1-or-more", respectively
- */
-case class ParameterCommandPart(wdlType: WdlType, name: String,
-                                prefix: Option[String], attributes: Map[String, String],
-                                postfixQuantifier: Option[String] = None) extends CommandPart {
-  override def toString: String = "${" + s"${wdlType.toWdlString} $name" + "}"
-
-  def instantiate(parameters: Map[String, WdlValue]): String = {
-    def wasDefaultValueUsed: Boolean = parameters.get(name).isEmpty && attributes.contains("default")
-
-    /* Order DOES matter in this match clause */
-    val paramValue = parameters.get(name) match {
-      case Some(value) => value
-      case None if attributes.contains("default") => WdlString(attributes.get("default").get)
-      case None if postfixQuantifier.isDefined && ParameterCommandPart.OptionalPostfixQuantifiers.contains(postfixQuantifier.head) => WdlString("")
-      case _ => throw new UnsupportedOperationException(s"Parameter $name not found")
-    }
-    val paramValueEvaluated = paramValue match {
-      case e: WdlExpression =>
-        /* TODO: this should never happen because expressions will be evaluated
-           before this method is called.  The reason why we can't always do it
-           here is because unless scope information is stored in the WdlExpression
-           itself (along with lookup and WdlFunctions objects), we'd have no way to
-           evaluate this with any reasonable variable dereferencing.
-         */
-        throw new UnsupportedOperationException("All WdlExpressions must be evaluated before calling this")
-      case x => x
-    }
-
-    if (!wasDefaultValueUsed && postfixQuantifier.isEmpty && wdlType != paramValueEvaluated.wdlType) {
-      throw new UnsupportedOperationException(s"Incompatible type for $name: need a $wdlType, got a ${paramValue.wdlType}")
+case class ParameterCommandPart(attributes: Map[String, String], expression: WdlExpression) extends CommandPart {
+  def attributesToString: String = if (attributes.nonEmpty) s"${attributes.map{case (k,v) => s"$k='$v'"}.mkString(", ")} " else ""
+  override def toString: String = "${" + s"$attributesToString${expression.toWdlString}" + "}"
+
+  def instantiate(declarations: Seq[Declaration], parameters: Map[String, WdlValue], functions: WdlFunctions = new NoFunctions): String = {
+    case class VariableNotFoundException(variable: String) extends RuntimeException(s"Could not find variable: $variable")
+    def lookup(key: String) = parameters.getOrElse(key, throw new VariableNotFoundException(key))
+
+    val value = expression.evaluate(lookup, functions) match {
+      case Success(v) => v
+      case Failure(f) => f match {
+        case v:VariableNotFoundException => declarations.find{_.name == v.variable} match {
```

Comment:
Do we really need this many layers of matches in a single closure?

Also, either `declarations.find(...)` or `declarations find { ... }`